### ISSUE TITLE
[Windows] terminal tab crashes when msys2 is on the PATH

### ISSUE THREAD

**Issue by jeroen** [rstudio/rstudio#6696]
In order for Windows users to build packages from source with rtools40, we need to put `make`, `bash`, etc on the PATH. We recommend people to do this in their `~/.Renviron` file:

```r
PATH="${RTOOLS40_HOME}\usr\bin;${PATH}"
```

However doing so will crash the terminal tab in rstudio due to dll conflicts. 

The problem is that bash should always have itself and other coreutils in the front of the path, but rstudio invokes bash.exe from git4windows but appends `/bin` all the way at the end of the PATH instead of the front.

As a result, when we invoke e.g. `ls` we get `ls.exe` from rtools40 while running inside bash.exe from git4windows. Sadly, mixing different msys2 / cygwin runtimes in a process will 💣. 

You can confirm by putting `/usr/bin` (or just `/bin`) on the front, then the problem disappears:

```sh
PATH="/usr/bin:$PATH"
```

So RStudio should probably always do automatically to make sure that other utilities are called from the same bash runtime.

**Comment by ronblum** [comment]
@jeroen Thank you for letting us know! This is especially helpful as we work on testing with R 4.0.0 beta. We'll review as we continue development of RStudio.

**Comment by jeroen** [comment]
Can you test with R 4.0.0 RC because we slipped in some last minute changes!

**Comment by gtritchie** [comment]
Note to self: wonder if this is the cause of issues like #6297

**Comment by jeroen** [comment]
R 4.0 is released now. Let me know if you need help with this.

**Comment by jeroen** [comment]
The solution from [here](https://github.com/rstudio/rstudio/issues/6809#issuecomment-624203688) is very simple: prepend `/usr/bin` to the path when starting bash.

```
PATH="/usr/bin:$PATH"
```

That way the bash invokes coreutils from the same cygwin/msys bundle as itself, and no conflict arises.

**Comment by gtritchie** [comment]
@jeroen Thanks, haven't had a chance to look into this yet; hopefully soon!

**Comment by gtritchie** [comment]
Fixed via #6828 (this fix is in 1.3-patch, and will be in master/1.4 shortly)

**Comment by R-Hannibal** [comment]
Hey @jeroen, thanks again for finding and reporting this issue. We have a fix in place in our latest preview build (available [HERE](https://rstudio.com/products/rstudio/download/preview/)).

If you get the chance will you let us know if the issue is resolved for you in that build (saw that you commented the issue was fixed on https://github.com/r-windows/rtools-installer/issues/11, but in theory this issue is not fixed in our [live 1.3 release](https://rstudio.com/products/rstudio/download/), and would only show as fixed in our new 1.3 patch build that first became available on **06/01/2020** on the preview page, linked above).

**Comment by R-Hannibal** [comment]
Adding the verified label here based off of your comment on [rtools installer issue](https://github.com/r-windows/rtools-installer/issues/11) @jeroen (I think I misread the dates originally, apologies).

Please let us know if you notice any re-occurrences of the issue and thank you for helping us to improve the IDE.
